llvm/test/CodeGen/AMDGPU/attributor-noalias-addrspace.ll

```diff
@@ -0,0 +1,596 @@
+; CHECK-NEXT:    [[ATOMICRMW_UMAX:%.*]] = atomicrmw umax ptr [[PTR5]], i32 21 syncscope("singlethread") monotonic, align 4, !noalias.addrspace [[META0]]
+; CHECK-NEXT:    [[ATOMICRMW_UMIN:%.*]] = atomicrmw volatile umin ptr [[PTR5]], i32 22 syncscope("singlethread") monotonic, align 4, !noalias.addrspace [[META0]]
+; CHECK-NEXT:    ret void
+;
+  br i1 %cond1, label %bb.1.true, label %bb.1.false
+bb.1.true:
+  br label %bb.1.end
+
+bb.1.false:
+  br label %bb.1.end
+
+bb.1.end:
+  %ptr4 = phi ptr [ %ptr1, %bb.1.true ], [ %ptr2, %bb.1.false ]
+  br i1 %cond2, label %bb.2.true, label %bb.2.end
+
+bb.2.true:
+  br label %bb.2.end
+
+bb.2.end:
+  %ptr5 = phi ptr [ %ptr4, %bb.1.end ], [ %ptr3, %bb.2.true ]
+  store i32 %val, ptr %ptr5
+  %atomicrmw.xchg = atomicrmw xchg ptr %ptr5, i32 12 monotonic, align 4
+  %atomicrmw.add = atomicrmw add ptr %ptr5, i32 13 monotonic, align 4
+  %atomicrmw.sub = atomicrmw sub ptr %ptr5, i32 14 monotonic, align 4
+  %atomicrmw.and = atomicrmw and ptr %ptr5, i32 15 monotonic, align 4
+  %atomicrmw.nand = atomicrmw nand ptr %ptr5, i32 16 monotonic, align 4
+  %atomicrmw.or = atomicrmw or ptr %ptr5, i32 17 monotonic, align 4
+  %atomicrmw.xor = atomicrmw xor ptr %ptr5, i32 18 monotonic, align 4
+  %atomicrmw.max = atomicrmw max ptr %ptr5, i32 19 monotonic, align 4
+  %atomicrmw.min = atomicrmw volatile min ptr %ptr5, i32 20 monotonic, align 4
+  %atomicrmw.umax = atomicrmw umax ptr %ptr5, i32 21 syncscope("singlethread") monotonic, align 4
+  %atomicrmw.umin = atomicrmw volatile umin ptr %ptr5, i32 22 syncscope("singlethread") monotonic, align 4
+  ret void
+}
+
+define amdgpu_kernel void @kernal_call_func(i1 %cond1, i1 %cond2, i32 %val) #0 {
+; CHECK-LABEL: define amdgpu_kernel void @kernal_call_func(
+; CHECK-SAME: i1 [[COND1:%.*]], i1 [[COND2:%.*]], i32 [[VAL:%.*]]) #[[ATTR2:[0-9]+]] {
+; CHECK-NEXT:    [[LPTR:%.*]] = alloca i32, align 4, addrspace(5)
+; CHECK-NEXT:    call void @callee_no_alias_addr_space_select(ptr addrspacecast (ptr addrspace(1) @gptr to ptr), ptr addrspacecast (ptr addrspace(5) @gptr2 to ptr), ptr addrspacecast (ptr addrspace(3) @gptr3 to ptr), i1 [[COND1]], i1 [[COND2]], i32 [[VAL]])
+; CHECK-NEXT:    call void @callee_alias_addr_space_branch(ptr addrspacecast (ptr addrspace(1) @gptr to ptr), ptr addrspacecast (ptr addrspace(5) @gptr2 to ptr), ptr addrspacecast (ptr addrspace(3) @gptr3 to ptr), i1 [[COND1]], i1 [[COND2]], i32 [[VAL]])
+; CHECK-NEXT:    ret void
+;
+  %lptr = alloca i32, align 4, addrspace(5)
+  %a = addrspacecast ptr addrspace(1) @gptr to ptr
+  %b = addrspacecast ptr addrspace(5) @gptr2 to ptr
+  %c = addrspacecast ptr addrspace(3) @gptr3 to ptr
+  call void @callee_no_alias_addr_space_select(ptr %a, ptr %b, ptr %c, i1 %cond1, i1 %cond2, i32 %val)
+  call void @callee_alias_addr_space_branch(ptr %a, ptr %b, ptr %c, i1 %cond1, i1 %cond2, i32 %val)
+  ret void
+}
+
+;.
+; CHECK: [[META0]] = !{i32 2, i32 3, i32 4, i32 5, i32 6, i32 10}
```

Comment:
Only one metadata? This looks like everything just in AS1. Can you add another tests where they are in AS5?